frontend/src/main/scala/bloop/exec/JvmProcessForker.scala

```diff
@@ -118,6 +121,11 @@ final class JvmForker(config: JdkConfig, classpath: Array[AbsolutePath]) extends
     val jvmOptions = jargs.map(_.stripPrefix("-J")) ++ config.javaOptions
     val fullClasspath = classpath ++ extraClasspath
     val fullClasspathStr = fullClasspath.map(_.syntax).mkString(File.pathSeparator)
+    val runOpts = if (envVars.nonEmpty) opts.copy(env = CommonOptions.currentEnv) else opts
```

Comment:
So turns out if we don't copy the env variable will also be set for the next run, which popped up as a failure in an unrelated test.

Reply:
This comment looks a bit weird, why is this caused? Environment variables are set on a session based basis, so different clients should have different environment variables.

Also, what variables are picked up in a BSP session?

Reply:
I can't remember why this solved the issue, but the problem was that setting environment variable to opts cause those variable to be available on the next run too. I will dig in to see why this fixed the tests.

Reply:
So `opts.env` was actually shared between tests (not sure where it's defined for them) and if I added an environment variable in one test, it was available in another.

`opts.copy(env = CommonOptions.currentEnv) ` helped, although this might not be the full solution. We should most likely have something like `opts.copy(env = CommonOptions.newEnv)`

All this could be an issue only in tests, but is JVMProcessForker invoked in a separate JVM for each session? If not then we have to copy the current environment instead o relying on lazy val `CommonOptions.currentEnv` What do you think @jvican  ?

Reply:
Thanks for the explanation and sorry for my delay in getting back to you @tgodzik `JvmProcessForker` is invoked in the same class loader so we should not be copying the environment for it, it will pick up whatever the environment is at a given moment. I would try to fix the issue with environment variables be shared in tests and remove this change. What do you think?

Reply:
Good idea! I was actually thinking about the same, but didn't manage to get back to it yet.  I will try to do it as soon as possible.